### ISSUE TITLE
An independent test set is needed

### ISSUE THREAD

**Issue by tom9358** [tom9358/nllb-tryout#17]
This issue requires some thinking before coding..

Currently, validation always happens as follows: we import whatever data, and split so that 2% is not used for training and is instead a validation set. This makes validation results between models with different datasets not really comparable. A model might appear to perform better simply because it was tested on an "easier" 2% of data rather than having superior predictive power.

That's the validation story. For test set, the story is: there is no test set! (but it doesn't per se have to exist yet, so let's focus on validation.)

We could create a fixed dataset - but there's a kind of tradeoff there: improved comparability, but if future data is of higher quality, then scores on this set are less representative of actual performance. On the other hand, currently we have a sort of moving target (or rather one with high variance and that is super sensitive on training data).

One option is to just download the current data and set apart 2%. Then we just remove those sentences from any future training set.

It might also be wise to have a validation set per data source / per domain, because tatoeba are usually very short sentences with relatively simple grammar and vocabulary. model is very likely to score very differently (a lot worse I expect) on other sets. Same goes for slightly less literal translations.

Maybe this is a hack?: All tatoeba sentences have unique identifiers, so - assuming that old ones do not change, which they **do**, but only rarely - we could say, we take every id that ends in 00 - 19. That's 2% of the data. :D just need to come up with something similar for other data sources.
